### DISCUSSION TITLE
S3 locks and expiry

### DISCUSSION THREAD

**Discussion by tcurdt** [k8up-io/k8up#975] [Q&A]
Not sure if this is more of a Restic or how k8up handles this.

I would like to S3 with versioning, expiry and object locks.
The idea is that with the backup credentials, the (encrypted) backups cannot be deleted - but they just expire after 30 days.

Is that supported?

**Comment by Kidswiss** [comment]
Hi @tcurdt 

Yes Restic would need to support this. Neither locking nor expiry are supported by Restic.

If you use expiry on a bucket containing Restic backups, it will probably corrupt the backups.

The self-hosted rest-server has an `--append-only` mode. However I've never used it.
